testcases/OpTestMakedump.py

```diff
@@ -0,0 +1,136 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+'''
+1. Trigger crash
+2. Check for Vmcore captured or not
+3. Run makedump on the vmcore captured
+'''
+import time
+import unittest
+
+import OpTestConfiguration
+from common.Exceptions import KernelOOPS, KernelKdump
+
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class OpTestMakedump(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.c = self.cv_SYSTEM.console
+        self.util = conf.util
+
+    def kernel_crash(self):
+        '''
+        This function will test the kdump followed by system
+        reboot. it has below steps
+        1. Take backup of files under /var/crash.
+        2. Trigger kernel crash: ``echo c > /proc/sysrq-trigger``
+        '''
+        self.c.run_command("mkdir -p /var/crash_bkp/")
+        if len(self.c.run_command("ls -1 /var/crash")) != 0:
+            self.c.run_command("mv /var/crash/* /var/crash_bkp/")
+            self.util.wait_for(self.c.run_command("sync"), 60)
+        else:
+            log.info("/var/crash/ directory is empty")
+        self.c.run_command("echo 1 > /proc/sys/kernel/sysrq")
+        try:
+            self.c.run_command("echo c > /proc/sysrq-trigger")
+        except (KernelOOPS, KernelKdump):
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+
+
+class Makedump(OpTestMakedump):
+    '''
+    this function will test the vmcore is captured or not
+    and run the makedump
+    '''
+
+    def check_run(self, cmd, condition):
+        res = self.c.run_command(cmd)
+        for value in res:
+            if condition in value:
+                log.info("command %s works well" % cmd)
+                return
+        self.fail("commnd %s failed" % cmd)
+
+    def makedump_check(self):
+        res = self.c.run_command("ls -1 /var/crash/")
+        crash_dir = self.c.run_command("cd /var/crash/%s" % res[0])
+        res = self.c.run_command("ls")
+        if 'vmcore' not in res[0]:
+            self.fail("vmcore is not saved")
+        else:
+            log.info("vmcore is saved")
+        self.c.run_command("makedumpfile -v")
+        self.check_run("makedumpfile --split -d 31 -l vmcore dump3 dump4",
+                       "The dumpfiles are saved to dump3, and dump4")
+        self.check_run("makedumpfile --reassemble dump3 dump4 dump5",
+                       "The dumpfile is saved to dump5.")
```

Comment:
is full stop necessary here. for other checks I dont see it.

Reply:
it just string check so not much matter i feel , if you think need to added  will respin the patch